File: packages/schemas/src/constitution-suggestion.ts
/** Constitution Suggestion types matching backend schemas. */

export type SuggestionStatus = "pending" | "accepted" | "rejected" | "expired";

export interface ConstitutionSuggestion {
  id: string;
  project_id: string;
  title: string;
  rationale: string | null;
  suggested_text: string;
  category: string | null;
  status: SuggestionStatus;
  source_metadata: Record<string, unknown> | null;
  created_at: string;
  updated_at: string;
}

export interface ConstitutionSuggestionList {
  items: ConstitutionSuggestion[];
  total: number;
}

export interface ConstitutionSuggestionResolve {
  action: "accept" | "reject";
}

File: apps/api/app/schemas/constitution_suggestion.py
"""FM-117: Constitution Suggestion schemas."""

import uuid
from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field

from app.models.constitution_suggestion import SuggestionStatus


class ConstitutionSuggestionRead(BaseModel):
    id: uuid.UUID
    project_id: uuid.UUID
    title: str
    rationale: str
    suggested_text: str
    category: str
    status: SuggestionStatus
    source_metadata: dict[str, Any] | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class ConstitutionSuggestionList(BaseModel):
    items: list[ConstitutionSuggestionRead]
    total: int


class ConstitutionSuggestionResolve(BaseModel):
    """Body to accept or reject a suggestion."""

    action: str = Field(..., pattern=r"^(accept|reject)$")
